osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -44,33 +58,23 @@ public class Catcher : Container, IKeyBindingHandler<CatchAction>
         /// </summary>
         private const float allowed_catch_range = 0.8f;
 
-        protected bool Dashing
+        /// <summary>
+        /// The drawable catcher for <see cref="CurrentState"/>.
+        /// </summary>
+        internal Drawable CurrentDrawableCatcher => currentCatcher.Drawable;
+
+        private bool dashing;
+
+        public bool Dashing
         {
             get => dashing;
-            set
+            protected set
             {
                 if (value == dashing) return;
 
                 dashing = value;
 
-                Trail |= dashing;
-            }
-        }
-
-        /// <summary>
-        /// Activate or deactivate the trail. Will be automatically deactivated when conditions to keep the trail displayed are no longer met.
-        /// </summary>
-        protected bool Trail
-        {
-            get => trail;
-            set
-            {
-                if (value == trail || AdditiveTarget == null) return;
-
-                trail = value;
-
-                if (Trail)
-                    beginTrail();
+                trails.DisplayTrail = value || HyperDashing;
```

Comment:
this is horrible logic. you can't arbitrarily or in a value in a set property. the chances of this breaking are 99.9% (it is not going to be applied unless `Dashing` is set, but there's a precondition equality check potentially causing the set to be a no-op).

Reply:
Would you mind having a look at the commits I just pushed in here and seeing if it improves things? I'm not 100% sure it does but after looking at this code for the better part of an hour I decided to avoid any bigger changes for now.